index.js

```diff
@@ -1,5 +1,13 @@
 'use strict';
 
+function getVaryHeaderPrefix(headers) {
+  const vary = headers.vary || headers.Vary;
+  if (vary && vary.search(/origin/i) === -1 && vary.indexOf('*') === -1) {
```

Comment:
maybe we should ensure the logic is more completed like this?

```js
vary = vary.split(/\s+,\s+/);
if (vary.any(item => item === '*' || item.match(/^origin$/i)) 
```

Reply:
Another option is the [`vary` npm module](https://github.com/jshttp/vary#readme), where it would be `return vary.append((headers.vary || headers.Vary || ''), 'Origin')`.

At the very least `.split(/\s+,\s+/)` will lead to a ReDOS that was getting tagged on lots of npm modules not too long ago, so likely want to use a different method to split the header apart if going the roll your own route.

Reply:
the vary npm module is more suitable, 👍